src/handlers/TelegramHandler.ts

```diff
@@ -0,0 +1,154 @@
+import base64url from "base64url";
+import deepmerge from "deepmerge";
+
+import { UX_MODE } from "../utils/enums";
+import { broadcastChannelOptions, getTimeout } from "../utils/helpers";
+import log from "../utils/loglevel";
+import PopupHandler from "../utils/PopupHandler";
+import AbstractLoginHandler from "./AbstractLoginHandler";
+import { CreateHandlerParams, LoginWindowResponse, TorusVerifierResponse } from "./interfaces";
+
+type PopupResponse = {
+  result: {
+    auth_date: number;
+    first_name: string;
+    hash: string;
+    id: number;
+    last_name: string;
+    photo_url: string;
+    username: string;
+  };
+  origin: string;
+  event: string;
+};
+
+export default class TelegramHandler extends AbstractLoginHandler {
+  private readonly RESPONSE_TYPE: string = "token";
+
+  private readonly SCOPE: string = "profile";
+
+  private readonly PROMPT: string = "select_account";
+
+  constructor(params: CreateHandlerParams) {
+    super(params);
+    this.setFinalUrl();
+  }
+
+  setFinalUrl(): void {
+    const finalUrl = new URL("https://oauth.telegram.org/auth");
```

Comment:
should we allow developers to pass a custom telegram url ? Do you think there could be a use case for it?